modules/services/utils/src/main/java/au/com/shiftyjelly/pocketcasts/utils/StorageUtil.kt

```diff
@@ -0,0 +1,14 @@
+package au.com.shiftyjelly.pocketcasts.utils
+
+import android.os.Environment
+import android.os.StatFs
+
+private const val lowStorageThreshold = 0.10
+
+fun isDeviceRunningOnLowStorage(): Boolean {
+    val statFs = StatFs(Environment.getExternalStorageDirectory().path)
+    val totalStorage = statFs.blockCountLong * statFs.blockSizeLong
+    val availableStorage = statFs.availableBlocksLong * statFs.blockSizeLong
+
+    return availableStorage < totalStorage * lowStorageThreshold
```

Comment:
```suggestion
    try {
        val statFs = StatFs(Environment.getExternalStorageDirectory().path)
        val totalStorage = statFs.blockCountLong * statFs.blockSizeLong
        val availableStorage = statFs.availableBlocksLong * statFs.blockSizeLong
        return availableStorage < totalStorage * lowStorageThreshold
    } catch (e: Exception) {
        Timber.e(e, "Error checking if low storage")
        return false
    }
```
I wonder if catching any exceptions here would be a good idea.

Reply:
It probably shouldn't be executed on the main thread since it calls`statvfs(2)`.